nova/src/circuits/supernova/mod.rs

```diff
@@ -501,33 +528,44 @@ pub(crate) mod tests {
     pub struct TestCircuit<F: Field>(PhantomData<F>);
 
     impl<F: PrimeField> NonUniformCircuit<F> for TestCircuit<F> {
-        const ARITY: usize = 1;
+        const ARITY: usize = 2;
 
         const NUM_CIRCUITS: usize = 2;
 
+        fn compute_selector(
+            &self,
+            _: ConstraintSystemRef<F>,
+            _: &FpVar<F>,
+            z: &[FpVar<F>],
+        ) -> Result<FpVar<F>, SynthesisError> {
+            // store selector in the first input var.
+            Ok(z[0].clone())
+        }
+
         fn generate_constraints(
             &self,
             _: ConstraintSystemRef<F>,
-            pc: &FpVar<F>,
-            pc_value: u64,
+            pc: u64,
             _: &FpVar<F>,
             z: &[FpVar<F>],
-        ) -> Result<(FpVar<F>, Vec<FpVar<F>>), SynthesisError> {
+        ) -> Result<Vec<FpVar<F>>, SynthesisError> {
             // alternate between + 2 and * 2
-            assert_eq!(z.len(), 1);
+            assert_eq!(z.len(), 2);
 
-            let pc_is_zero = pc.is_zero()?;
-            let pc_next = pc_is_zero.select(&FpVar::one(), &FpVar::zero())?;
-
-            let x = &z[0];
+            let x = &z[1];
 
-            let y = match pc_value {
+            let y = match pc {
```

Comment:
this looks a bit like collatz... :)

Reply:
it was renamed in the trait definition :man_shrugging: